crypto/x509/v3_alt.c

```diff
@@ -185,24 +184,14 @@ STACK_OF(CONF_VALUE) *i2v_GENERAL_NAME(X509V3_EXT_METHOD *method,
     case GEN_IPADD:
         p = gen->d.ip->data;
         if (gen->d.ip->length == 4)
-            BIO_snprintf(oline, sizeof(oline), "%d.%d.%d.%d",
-                         p[0], p[1], p[2], p[3]);
-        else if (gen->d.ip->length == 16) {
-            oline[0] = 0;
-            for (i = 0; i < 8; i++) {
-                BIO_snprintf(htmp, sizeof(htmp), "%X", p[0] << 8 | p[1]);
-                p += 2;
-                strcat(oline, htmp);
-                if (i != 7)
-                    strcat(oline, ":");
-            }
-        } else {
-            if (!X509V3_add_value("IP Address", "<invalid>", &ret))
-                return NULL;
-            break;
-        }
-        if (!X509V3_add_value("IP Address", oline, &ret))
-            return NULL;
+            tmp = ipv4_to_asc(p);
+        else if (gen->d.ip->length == 16)
+            tmp = ipv6_to_asc(p);
+        else
+            tmp = OPENSSL_strdup("<invalid>");
```

Comment:
maybe print the length received?

Reply:
Good thought. This motivated me to do a further simplification:
Replace `ipv4_to_asc()` and `ipv6_to_asc()` by `ipaddr_to_asc()` including len error output.

Reply:
Note that for `print_nc_ipadd()`, which should print two consecutive IP addresses of the same version (v4 or v6), I tried to split the length as evenly as possible (covering in a sense also the error case that they are in total not 2 * 4 or 2 * 16 bytes long).